### DISCUSSION TITLE
How to avoid dropping word-initial and word-final apostrophes?

### DISCUSSION THREAD

**Discussion by Willem3141** [espeak-ng/espeak-ng#1804] [Q&A] [answered]
I'm trying to make a voice for the conlang [Na'vi](https://en.wikipedia.org/wiki/Na'vi_language). One peculiarity of Na'vi is that it uses apostrophes for glottal stops, and they can occur also in the beginning and end of words. For example, **'eylan** and **eylan** are distinct words that should be pronounced differently: one with a clear glottal stop, and the other without.

Sadly espeak-ng is dropping these apostrophes by treating them as punctuation. I've already discovered the setting `punct_within_word` and set it to `{ '\'', 0 }` in `tr_languages.c`. This works for apostrophes in the middle of the word, but not for ones in the beginning and end. I think this is because the code [checks if the previous character is alphanumeric](https://github.com/espeak-ng/espeak-ng/blob/master/src/libespeak-ng/translate.c#L1177).

Is there another way to get espeak-ng not to drop these? There must be other languages that have this problem, right?

**Comment by jaacoppi** [comment, marked as answer]
I don't think there's any option at the moment.

If you write a patch to implement it, make it a language options in langopts.c instead of tr_languages.c
